Assets/MixedRealityToolkit/UX/Scripts/Tooltip/ToolTipConnector.cs

```diff
@@ -0,0 +1,258 @@
+﻿//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+//
+using UnityEngine;
+
+namespace MixedRealityToolkit.UX.ToolTips
+{
+    /// <summary>
+    /// Connects a ToolTip to a target
+    /// Maintains that connection even if the target moves
+    /// </summary>
+    public class ToolTipConnector : MonoBehaviour
+    {
+        public enum FollowTypeEnum
+        {
+            AnchorOnly,             // The anchor will follow the target - pivot remains unaffected
+            PositionOnly,           // Anchor and pivot will follow target position, but not rotation
+            PositionAndYRotation,   // Anchor and pivot will follow target like it's parented, but only on Y axis
+            PositionAndRotation,    // Anchor and pivot will follow target like it's parented
+        }
+
+        public enum OrientTypeEnum
+        {
+            OrientToObject,     // Tooltip will maintain anchor-pivot relationship relative to target object
+            OrientToCamera,     // Tooltip will maintain anchor-pivot relationship relative to camera
+        }
+
+        public enum PivotModeEnum
+        {
+            Manual,         // Tooltip pivot will be set manually
+            Automatic,      // Tooltip pivot will be set relative to object/camera based on specified direction and line length
+        }
+
+        public enum PivotDirectionEnum
+        {
+            Manual,         // Direction will be specified manually
+            North,
+            NorthEast,
+            East,
+            SouthEast,
+            South,
+            SouthWest,
+            West,
+            NorthWest,
+            InFront,
+        }
+        
+        public GameObject Target;
+
+        public FollowTypeEnum FollowType = FollowTypeEnum.AnchorOnly;
+        public PivotModeEnum PivotMode = PivotModeEnum.Manual;
+        public PivotDirectionEnum PivotDirection = PivotDirectionEnum.North;
+        public OrientTypeEnum PivotDirectionOrient = OrientTypeEnum.OrientToObject;
+        public Vector3 ManualPivotDirection = Vector3.up;
+        public Vector3 ManualPivotLocalPosition = Vector3.up;
+        [Range(0f, 2f)]
+        public float PivotDistance = 0.25f;
+
+        private void OnEnable()
+        {
+            if (!FindToolTip())
+                return;
+
+            ManualPivotLocalPosition = transform.InverseTransformPoint (toolTip.PivotPosition);
+        }
+
+        private bool FindToolTip()
+        {
+            if (toolTip == null)
+            {
+                toolTip = GetComponent<ToolTip>();
```

Comment:
if the `Tooltip` is required for this class and gameobject then be sure to add the required component attribute and use `gameObject.EnsureComponent();`